File: .tmp_py4013/batch_13_dump/3448_count_substrings_divisible_by_last_digit.js
// LeetCode 3448 - Count Substrings Divisible By Last Digit
// https://leetcode.com/problems/count-substrings-divisible-by-last-digit/

var countSubstrings = function(s) {
    let ans = 0;
    const n = s.length;
    for (let r = 0; r < n; r++) {
        const last = s.charCodeAt(r) - 48;
        if (last === 0) continue;
        let mod = 0;
        let p = 1 % last;
        for (let l = r; l >= 0; l--) {
            mod = (mod + (s.charCodeAt(l) - 48) * p) % last;
            p = (p * 10) % last;
            if (mod === 0) ans++;
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check17/3448_count_substrings_divisible_by_last_digit.rs
struct Solution;
// LeetCode 3448 - Count Substrings Divisible By Last Digit
// https://leetcode.com/problems/count-substrings-divisible-by-last-digit/

impl Solution {
    pub fn count_substrings(s: String) -> i64 {
        let bytes = s.as_bytes();
        let n = bytes.len();
        let mut ans = 0i64;
        for r in 0..n {
            let last = (bytes[r] - b'0') as i32;
            if last == 0 {
                continue;
            }
            let mut modulus = 0;
            let mut p = 1 % last;
            for l in (0..=r).rev() {
                modulus = (modulus + (bytes[l] - b'0') as i32 * p) % last;
                p = (p * 10) % last;
                if modulus == 0 {
                    ans += 1;
                }
            }
        }
        ans
    }
}

fn main() {}
